### PR TITLE
Add missing request error logging

### PR DESCRIPTION
**Pull request by MatkovIvan**
Log a real error instead of
`HTTP response received with status code=0 and payload=(null)`

### CODE REVIEW

File: AppCenter/AppCenter/Internals/Ingestion/MSHttpIngestion.m [hunk 1]
```diff
@@ -258,48 +259,22 @@ - (void)sendCallAsync:(MSIngestionCall *)call {
...
-                    // NSJSONSerialization escapes paths by default so we
-                    // replace them.
-                    payload =
-                        [[[NSString alloc] initWithData:jsonData
-                                               encoding:NSUTF8StringEncoding]
-                            stringByReplacingOccurrencesOfString:@"\\/"
-                                                      withString:@"/"];
-                  }
-                }
+              // Don't loose time pretty printing if not going to be printed.
```
Comment (achocron) [inline AppCenter/AppCenter/Internals/Ingestion/MSHttpIngestion.m]: nit: loose -> lose

File: AppCenter/AppCenterTests/MSUtilityTests.m [hunk 2]
```diff
@@ -198,6 +198,23 @@ - (void)testCreateSha256 {
           @"6dd79f2770a0bb38073b814a5ff000647b37be5abbde71ec9176c6ce0cb32a27"]);
 }
 
+- (void)testPrettyPrintJson {
```
Comment (achocron) [inline AppCenter/AppCenterTests/MSUtilityTests.m]: Can you split these into three tests? Seems like three cases are tested at once

**Comment by guperrot** [comment]
We should update changelog at same time we do PRs.

**Comment by TroubleMakerBen** [comment]
I agree @guperrot, as the console logs are public facing.

**Pull request merged**